File: LeetCodeSolutions/Problem475/Heaters.java
class Heaters {
    public int findRadius(int[] houses, int[] heaters) {
        
        Arrays.sort(heaters);
        Arrays.sort(houses);
        
        int ans = 0;
        
        if(heaters[0]>houses[0])
        ans = Math.max(ans,heaters[0]-houses[0]);
        
        if(heaters[heaters.length-1]<houses[houses.length-1])
        ans = Math.max(ans,houses[houses.length-1]-heaters[heaters.length-1]);
        
        // System.out.println(ans);
        for(int i = 0,j=0; i < houses.length;i++ )
        {
            for(;j+1<heaters.length;j++)
            {
                if(Math.abs(heaters[j+1] - houses[i])>Math.abs(heaters[j] - houses[i]))
                    break;
            }
            ans = Math.max(ans,Math.abs(heaters[j] - houses[i]));
        }
        return ans;
    }

File: LeetCodeSolutions/Problem475/Heaters.cpp
#include <vector>
#include <algorithm>
namespace leetcode
{
    using namespace std;
	class problem475
	{
	public:
        int findRadius(vector<int>& houses, vector<int>& heaters)
        {
            sort(houses.begin(), houses.end());
            sort(heaters.begin(), heaters.end());


            long long res = INT_MIN;
            heaters.insert(heaters.begin(), INT_MIN);
            heaters.push_back(INT_MAX);

            for (auto i : houses)
            {
                auto it = lower_bound(heaters.begin(), heaters.end(), i);
                res = max(res, *it == i ? 0 : min((long long)i - (long long)*(it - 1), (long long)*it - (long long)i));
            }

            return res;
        }
	};
}